cinema/serializers.py

```diff
@@ -68,13 +79,61 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available"
         )
 
 
+class TicketSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+
+    def validate(self, attrs):
+        Ticket.validate_seat_and_row(
+            attrs["seat"],
+            attrs["movie_session"].cinema_hall.seats_in_row,
+            attrs["row"],
+            attrs["movie_session"].cinema_hall.rows,
+            serializers.ValidationError,
+        )
+        return attrs
+
+
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = SerializerMethodField()
```

Comment:
You can use nested serializer here - create TicketSeatsSerializer with row / seat fields and pass it to taken_places